### ISSUE TITLE
PHP fatal errors when run

### ISSUE THREAD

**Issue by nask0** [sensiolabs-de/deprecation-detector#89]
git clone <deprecation-detector-git-url>
composer install
box build
./deprecation-detector.phar 

```
Checking your application for deprecations - this could take a while ...

Loading RuleSet...

PHP Catchable fatal error:  Argument 2 passed to SensioLabs\DeprecationDetector\RuleSet\DirectoryTraverser::traverse() must be an instance of SensioLabs\DeprecationDetector\RuleSet\RuleSet, boolean given, called in phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/RuleSet/Loader/ComposerLoader.php on line 115 and defined in phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/RuleSet/DirectoryTraverser.php on line 34
PHP Stack trace:
PHP   1. {main}() /var/www/html/deprecation-detector/deprecation-detector.phar:0
PHP   2. require() /var/www/html/deprecation-detector/deprecation-detector.phar:10
PHP   3. Symfony\Component\Console\Application->run() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/bin/deprecation-detector:17
PHP   4. Symfony\Component\Console\Application->doRun() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/vendor/symfony/console/Application.php:120
PHP   5. Symfony\Component\Console\Application->doRunCommand() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/vendor/symfony/console/Application.php:189
PHP   6. Symfony\Component\Console\Command\Command->run() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/vendor/symfony/console/Application.php:838
PHP   7. SensioLabs\DeprecationDetector\Console\Command\CheckCommand->execute() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/vendor/symfony/console/Command/Command.php:256
PHP   8. SensioLabs\DeprecationDetector\DeprecationDetector->checkForDeprecations() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/Console/Command/CheckCommand.php:121
PHP   9. SensioLabs\DeprecationDetector\RuleSet\Loader\ComposerLoader->loadRuleSet() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/DeprecationDetector.php:77
PHP  10. SensioLabs\DeprecationDetector\RuleSet\Loader\ComposerLoader->loadPackageRuleSet() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/RuleSet/Loader/ComposerLoader.php:50
PHP  11. SensioLabs\DeprecationDetector\RuleSet\DirectoryTraverser->traverse() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/RuleSet/Loader/ComposerLoader.php:115

Catchable fatal error: Argument 2 passed to SensioLabs\DeprecationDetector\RuleSet\DirectoryTraverser::traverse() must be an instance of SensioLabs\DeprecationDetector\RuleSet\RuleSet, boolean given, called in phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/RuleSet/Loader/ComposerLoader.php on line 115 and defined in phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/RuleSet/DirectoryTraverser.php on line 34

Call Stack:
    0.0132    1368160   1. {main}() /var/www/html/deprecation-detector/deprecation-detector.phar:0
    0.0133    1364088   2. require('phar:///var/www/html/deprecation-detector/deprecation-detector.phar/bin/deprecation-detector') /var/www/html/deprecation-detector/deprecation-detector.phar:10
    0.0197    2858304   3. Symfony\Component\Console\Application->run() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/bin/deprecation-detector:17
    0.0212    3189728   4. Symfony\Component\Console\Application->doRun() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/vendor/symfony/console/Application.php:120
    0.0214    3230152   5. Symfony\Component\Console\Application->doRunCommand() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/vendor/symfony/console/Application.php:189
    0.0214    3230728   6. Symfony\Component\Console\Command\Command->run() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/vendor/symfony/console/Application.php:838
    0.0216    3234792   7. SensioLabs\DeprecationDetector\Console\Command\CheckCommand->execute() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/vendor/symfony/console/Command/Command.php:256
    0.0352    7424736   8. SensioLabs\DeprecationDetector\DeprecationDetector->checkForDeprecations() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/Console/Command/CheckCommand.php:121
    0.0354    7464592   9. SensioLabs\DeprecationDetector\RuleSet\Loader\ComposerLoader->loadRuleSet() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/DeprecationDetector.php:77
    0.0372    7958200  10. SensioLabs\DeprecationDetector\RuleSet\Loader\ComposerLoader->loadPackageRuleSet() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/RuleSet/Loader/ComposerLoader.php:50
    0.0372    7958800  11. SensioLabs\DeprecationDetector\RuleSet\DirectoryTraverser->traverse() phar:///var/www/html/deprecation-detector/deprecation-detector.phar/src/RuleSet/Loader/ComposerLoader.php:115
```

php -v
```
PHP 5.6.15 (cli) (built: Oct 29 2015 14:59:23) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies
    with Xdebug v2.3.3, Copyright (c) 2002-2015, by Derick Rethan
```

**Comment by MarvinKlemp** [comment]
Looks like the code directories are not resolved correctly, if you use 
`deprecation-detector.phar check src/ vendor/` it should work fine.

**Comment by nask0** [comment]
Maybe i mess up something IDK.
I tried invoking it in this ways, with the same result as in first post:
1. /var/www/html/deprecation-detector/deprecation-detector.phar check /var/www/html/project/src/
2. /var/www/html/deprecation-detector/deprecation-detector.phar check ./src
3. /var/www/html/deprecation-detector/deprecation-detector.phar check src/

**Comment by MarvinKlemp** [comment]
No it's a bug 
but if you use `/var/www/html/deprecation-detector/deprecation-detector.phar check /var/www/html/project/src/ /var/www/html/project/vendor/` it should work

**Comment by haphan** [comment]
@MarvinKlemp Using absolute path does not work for me. This is a bug.

**Comment by nask0** [comment]
@haphan +1

**Comment by BallisticPain** [comment]
Definitely a :bug: ... however, when fixed using #92 merged locally, I run into another error. I'm posting it into this thread since the thread title is `PHP fatal errors when run`.

```
~/Development/project-upgrade-test (106_upgrade_symfony) $ ./../deprecation-detector/bin/deprecation-detector
Checking your application for deprecations - this could take a while ...

Loading RuleSet...

PHP Catchable fatal error:  Argument 1 passed to SensioLabs\DeprecationDetector\RuleSet\RuleSet::merge() must implement interface SensioLabs\DeprecationDetector\FileInfo\DeprecationCollectionInterface, null given, called in /Users/ballisticpain/Development/deprecation-detector/src/RuleSet/Loader/ComposerLoader.php on line 50 and defined in /Users/ballisticpain/Development/deprecation-detector/src/RuleSet/RuleSet.php on line 45

Catchable fatal error: Argument 1 passed to SensioLabs\DeprecationDetector\RuleSet\RuleSet::merge() must implement interface SensioLabs\DeprecationDetector\FileInfo\DeprecationCollectionInterface, null given, called in /Users/ballisticpain/Development/deprecation-detector/src/RuleSet/Loader/ComposerLoader.php on line 50 and defined in /Users/ballisticpain/Development/deprecation-detector/src/RuleSet/RuleSet.php on line 45
```

**Comment by MarvinKlemp** [comment]
@BallisticPain could you try #93 ?

**Comment by BallisticPain** [comment]
@MarvinKlemp Sure thing. :)

**Comment by BallisticPain** [comment]
Just to keep anyone having to run to #93. It did work for me.

Thanks!

**Comment by haphan** [comment]
#93 works for me. However, we probably need a test to cover this case.

**Comment by MarvinKlemp** [comment]
Fixes in #93
